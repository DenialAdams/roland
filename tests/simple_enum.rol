enum ReticulateSplines {
   Always,
   Sometimes,
   Never
}

proc process_data(reticulate_splines: ReticulateSplines) {
   if reticulate_splines == ReticulateSplines::Always {
      print("Reticulating splines...");
   } else if reticulate_splines == ReticulateSplines::Sometimes {
      print("Maybe reticulating splines...");
   }
}

proc main() {
   process_data(ReticulateSplines::Always);
   process_data(ReticulateSplines::Sometimes);
   process_data(ReticulateSplines::Never);
}
