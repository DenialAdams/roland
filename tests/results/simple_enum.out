Reticulating splines...
Maybe reticulate_splines...
